chain/jsonrpc-primitives/src/types/config.rs

```diff
@@ -0,0 +1,78 @@
+use crate::types::blocks::BlockReference;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Serialize, Deserialize)]
+pub struct RpcProtocolConfigRequest {
+    #[serde(flatten)]
+    pub block_reference: BlockReference,
+}
+
+impl RpcProtocolConfigRequest {
+    pub fn parse(
+        value: Option<Value>,
+    ) -> Result<RpcProtocolConfigRequest, crate::errors::RpcParseError> {
+        crate::utils::parse_params::<BlockReference>(value)
+            .map(|block_reference| RpcProtocolConfigRequest { block_reference })
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct RpcProtocolConfigResponse {
+    #[serde(flatten)]
+    pub config_view: near_chain_configs::ProtocolConfigView,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum RpcProtocolConfigError {
+    #[error("Block not found")]
+    BlockNotFound(String),
```

Comment:
We use `UnknownBlock` and `UnavailableBlock` to distinguish two possible `BlockNotFound` reasons, so I suggest we keep using it:

```suggestion
    #[error("Block has never been observed: {0}")]
    UnknownBlock(String),
```

@khorolets we need to go over the blocks implementation and fix it there (we already did it for chunks)